### PR TITLE
[CPU] Fix mixing vex and non-VEX instructions

### PR DESCRIPTION
**Pull request by EgorDuplensky**
### Details:
- Quote:
   The Skylake microarchitecture implements a different state
   machine than prior generations to manage the YMM state transition
   associated with mixing SSE and AVX instructions.
   It no longer saves the entire upper YMM state when executing
   an SSE instruction when in “Modified and Unsaved” state,
   but saves the upper bits of individual register.
   As a result, mixing SSE and AVX instructions will experience
   a penalty associated with partial register dependency of
   the destination registers being used and additional blend
   operation on the upper bits of the destination registers.

Such type of penalties have a huge impact on openvino's and oneDNN's kernels.
Basically the mixing of VEX and non-VEX instructions should be avoided.

### Tickets:
 - 54930
 - 61093
 - 38077

oneDNN PR:
- https://github.com/openvinotoolkit/oneDNN/pull/125

### CODE REVIEW

**Comment by EgorDuplensky** [comment]
NOTE: Should not be merged until verified on all the archs (Intel vector)

**Comment by ilya-lavrenov** [comment]
should not we merge to master first to perform extensive validation? I suppose 2021.4 has less testing..

**Comment by EgorDuplensky** [comment]
> Member

This Fix is already on master

**Comment by EgorDuplensky** [comment]
Update: the change was verified locally and needs to be verified with models on all the CPU architecture types.

**Comment by moslex** [comment]
@dmitry-gorokhov , could you please review. This is for 2021.4.3 LTS bugfix release.

**Comment by ilya-lavrenov** [comment]
is it still valid?

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.
